### PR TITLE
optional trailing port arguments ip4/ip6

### PR DESCRIPTION
**Pull request by adamyg**
ticket #18

### CODE REVIEW

**Comment by troglobit** [comment]
Hi and thank you for your contribution!  I've been auditing your changes to fix #18, and if I focus on the functional changes only, it solves the problem :smiley: 

However, there are lots of unrelated changes to the tree, e.g. good whitespace fixes and some unmotivated ones.  Each of these should either be a separate commit, or dropped from this patch series altogether.  Also, when submitting changes to an upstream project you should try to clean up the commits so they are clear and isolated, here you remove code from the first commit in the second due to copy-paste I guess.  You also need to write a summary of your changes with motivations.

Now, I can do a GitHub review of each of your changes and you can take this opportunity to learn a thing or two, but that'll take me a while to finish, so before I spend some of my weekend on it, I have to ask; are you interested in that type of feedback and willing to do the cleanup necessary?

**Comment by adamyg** [comment]
Hi, sure - we all work to a different set of rules, happy to experience alternatives. Shall admit commit a little messy, pushed from an internal repo and then realized it was an older branch which in-turn i then updated; possible rebase use-case; yet we never look at individual commits only the resulting artifact. As for white-space tweaks, no issues, stopped having these type of wars years ago; can refactor and/or clang-format.

**Comment by troglobit** [comment]
OK great, I'll send you a detailed review later today!

**Review by troglobit — CHANGES_REQUESTED** [review]
There, lots of minor details and a few major ones (`getuid()` in particular).

I think the easiest way to fix the mess with the two commits is to just to squash them when rebasing on the latest master.  But I leave that up to you to sort out, now that you're aware of it.

**Comment by adamyg** [comment]
pr comments applied

**Comment by troglobit** [comment]
Awesome, looks *much* better, thank you!  There are a few lingering things though that I'd like you to address before I merge, I've left those comments open from the first review, and updated a couple of them.

File: src/inetaddr.c
Comment (adamyg) [reply src/inetaddr.c]: original specialized geteuid() check restored: see main

File: src/mcjoin.c
Comment (adamyg) [reply src/mcjoin.c]: original source restored

**Comment by troglobit** [comment]
Great work on this, thank you for taking the time! :heart:

**Review by troglobit — APPROVED** [review]

**Pull request merged**
